pywr-core/src/aggregated_node.rs

```diff
@@ -605,4 +687,141 @@ mod tests {
+
+        output_node.set_cost(Some((-10.0).into()));
+
+        let output_node = network.get_mut_node_by_name("output", Some("1")).unwrap();
+        output_node.set_max_flow_constraint(Some(100.0.into())).unwrap();
+
+        output_node.set_cost(Some((-5.0).into()));
+
+        // Set-up assertion for "output-0" node
+        let idx = network.get_node_by_name("link", Some("0")).unwrap().index();
+        let expected = Array2::from_elem((366, 10), 100.0);
+        let recorder = AssertionRecorder::new("link-0-flow", MetricF64::NodeOutFlow(idx), expected, None, None);
+        network.add_recorder(Box::new(recorder)).unwrap();
+
+        // Set-up assertion for "output-1" node
+        let idx = network.get_node_by_name("link", Some("1")).unwrap().index();
+        let expected = Array2::from_elem((366, 10), 0.0);
+        let recorder = AssertionRecorder::new("link-1-flow", MetricF64::NodeOutFlow(idx), expected, None, None);
+        network.add_recorder(Box::new(recorder)).unwrap();
+
+        let model = Model::new(default_time_domain().into(), network);
+
+        run_all_solvers(&model, &["clp"], &[]);
+    }
+
+    /// Test double mutual exclusive flows
+    ///
+    /// The model has a single input that diverges to three links. Two sets of mutual exclusivity
+    /// constraints are defined, one for the first two links and one for the last two links. This
+    /// tests that a node can appear in two different mutual exclusivity constraints.
+    #[test]
+    fn test_double_mutual_exclusivity() {
+        let mut network = Network::default();
+
+        let input_node = network.add_input_node("input", None).unwrap();
+        let link_node0 = network.add_link_node("link", Some("0")).unwrap();
+        let output_node0 = network.add_output_node("output", Some("0")).unwrap();
+
+        network.connect_nodes(input_node, link_node0).unwrap();
+        network.connect_nodes(link_node0, output_node0).unwrap();
+
+        let link_node1 = network.add_link_node("link", Some("1")).unwrap();
+        let output_node1 = network.add_output_node("output", Some("1")).unwrap();
+
+        network.connect_nodes(input_node, link_node1).unwrap();
+        network.connect_nodes(link_node1, output_node1).unwrap();
+
+        let link_node2 = network.add_link_node("link", Some("2")).unwrap();
+        let output_node2 = network.add_output_node("output", Some("2")).unwrap();
+
+        network.connect_nodes(input_node, link_node2).unwrap();
+        network.connect_nodes(link_node2, output_node2).unwrap();
+
+        let _me_node = network.add_aggregated_node(
+            "mutual-exclusivity-01",
+            None,
+            &[vec![link_node0], vec![link_node1]],
+            Some(Relationship::new_exclusive(0, 1)),
+        );
+        let _me_node = network.add_aggregated_node(
+            "mutual-exclusivity-12",
+            None,
+            &[vec![link_node1], vec![link_node2]],
+            Some(Relationship::new_exclusive(0, 1)),
```

Comment:
I think this might be quite tricky without minimum flow support. It could activate something, but then not use it. However, it does beg the question whether we should be able to output the state of the binary variables.